sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -283,7 +283,7 @@ public void onFrameMetricCollected(
   }
 
   @SuppressLint("NewApi")
-  private @Nullable ProfilingTraceData onTransactionFinish(
+  private @Nullable synchronized ProfilingTraceData onTransactionFinish(
```

Comment:
This is called only in the other `onTransactionFinish`, which is already `synchronized`, so i'm not sure it makes a difference here, but it doesn't harm, either

Reply:
mm, not really, it's also called in `close()` and when scheduling the finish timeout, so likely one of those was nullifying the current transaction data